jdaviz/configs/specviz/plugins/unit_conversion/unit_conversion.py

```diff
@@ -159,47 +219,114 @@ def _on_spectral_unit_changed(self, *args):
-                self.flux_or_sb_selected = 'Surface Brightness'
-            else:
-                self.flux_or_sb_selected = 'Flux'
-
-    @observe('flux_or_sb_selected')
+    @observe('flux_or_sb_selected', 'flux_unit_selected', 'sb_unit_selected')
     def _translate(self, *args):
         # currently unsupported, can be supported with a scale factor
         if self.app.config == 'specviz':
             return
 
-        # Check for a scale factor/data passed through spectral extraction plugin.
-        specs_w_factor = [spec for spec in self.app.data_collection
-                          if "_pixel_scale_factor" in spec.meta]
-        # Translate if we have a scale factor
-        if specs_w_factor:
-            self.translate_units(self.flux_or_sb_selected)
-        # The translator dropdown hasn't been loaded yet so don't try translating
-        elif not self.show_translator:
-            return
-        # Notify the user to extract a spectrum before using the surface brightness/flux
-        # translation. Can be removed after all 1D spectra in Cubeviz pass through
-        # spectral extraction plugin (as the scale factor will then be stored).
-        else:
-            raise ValueError("No collapsed spectra in data collection, \
-                              please collapse a spectrum first.")
+        flux_or_sb = None
+
+        if args:
+            # need to check current y_unit to see if we need to translate
+            y_unit = self.spectrum_viewer.state.y_display_unit
+            for arg in args:
+                if arg['name'] == 'flux_unit_selected':
+                    # don't translate if self.flux_or_sb == Flux
+                    if not check_if_unit_is_per_solid_angle(y_unit):
+                        return
+                    flux_or_sb = 'Flux'
+                elif arg['name'] == 'sb_unit_selected':
+                    # don't translate if self.flux_or_sb == Surface Brightness
+                    if check_if_unit_is_per_solid_angle(y_unit):
+                        return
+                    flux_or_sb = 'Surface Brightness'
+                elif arg['name'] == 'flux_or_sb_selected':
+                    flux_or_sb = self.flux_or_sb_selected
+                else:
+                    return
+
+        # we want to raise an error if a user tries to translate with an
+        # untranslated Flux unit using the API
+        untranslatable_units = self.untranslatable_units()
+        untranslatable_units = units_to_strings(untranslatable_units)
+
+        if hasattr(self, 'flux_unit'):
+            if (self.flux_unit.selected in untranslatable_units) \
+                 and (flux_or_sb == 'Surface Brightness'):
+                raise ValueError(
+                    f"Selected flux unit is not translatable. Please choose a flux unit "
+                    f"that is not in the following list: {untranslatable_units}."
+                )
+
+        self.translate_units(flux_or_sb)
```

Comment:
I can add a conditional or raise an error if `flux_or_sb = None`. I can't see a case where it would be None or if an empty return would be triggered, but it would be nice to get some sort of message as opposed to just the return